library/alloc/src/rc.rs

```diff
@@ -403,7 +431,14 @@ impl<T> Rc<T> {
     /// impl Gadget {
     ///     /// Construct a reference counted Gadget.
     ///     fn new() -> Rc<Self> {
-    ///         Rc::new_cyclic(|me| Gadget { me: me.clone() })
+    ///         // `me` is an uninitialized `Weak` at the start of the closure.
+    ///         Rc::new_cyclic(|me| {
+    ///             // Create the actual struct here.
+    ///             Gadget { me: me.clone() }
+    ///
+    ///             // Before returning, `new_cyclic` fixes up the pointers
+    ///             // and reference counts inside gadget.
```

Comment:
What does this mean?

> fixes up the pointers and reference counts

It doesn't touch any pointers right? It only increments the strong counter of the Rc to 1.

> inside gadget

There's no variable named `gadget` here, so it's not entirely clear what this refers to. new_cyclic does not modify the `Gadget` struct itself.